client/container_list.go

```diff
@@ -35,7 +37,14 @@ func (cli *Client) ContainerList(ctx context.Context, options types.ContainerLis
 	}
 
 	if options.Filter.Len() > 0 {
-		filterJSON, err := filters.ToParam(options.Filter)
+		filterJSON := ""
+		err := errors.New("")
+		// for daemons older than v1.10, filter must be of the form map[string][]string
+		if cli.version != "" && version.Version(cli.version).LessThan(version.Version("1.10")) {
```

Comment:
> I wasn't sure we wanted to change it

I didn't say to change it, I'd prefer to add a new function to the filters package that handle it.

> how does Swarm indicate to the client what API version to use?

tbh, the fact that we're pushing this to the client sounds like a bad idea. Swarm should handle any transformation it needs to do to call to old daemons. It looks like you're trying to use the same client to talk with all daemons, I'm not sure it's the best approach.